src/useSnapEdgeToParent/helpers.ts

```diff
@@ -1,69 +1,77 @@
-import { RefObject } from 'react'
+export interface RefAndRectMap {
+  target: RefAndRect
+  parent: RefAndRect
+  container: RefAndRect
+}
+
+interface RefAndRect {
+  ref: (node: HTMLElement | null) => void
+  rect: ClientRect | DOMRect | null
+}
 
-export interface Refs<
-  Target extends HTMLElement = HTMLElement,
-  Parent extends HTMLElement = HTMLElement,
-  Container extends HTMLElement = HTMLElement
-  > {
-  target: RefObject<Target>
-  parent: RefObject<Parent>
-  container: RefObject<Container>
+export enum Vertical {
+  BOTTOM = 'top',
+  TOP = 'bottom',
+  AUTO = 'auto'
 }
 
-export type Vertical = 'top' | 'bottom'
-export type Horizontal = 'left' | 'right'
+export enum Horizontal {
+   LEFT = 'right',
+   RIGHT = 'left',
+   AUTO = 'auto'
+}
 
 export interface AutoPosition {
-  vertical?: Vertical | 'auto'
-  horizontal?: Horizontal | 'auto'
+  vertical?: Vertical
+  horizontal?: Horizontal
 }
```

Comment:
Maybe it's better to change these fields to required fields and use the Partial util type in the places where they are optional.